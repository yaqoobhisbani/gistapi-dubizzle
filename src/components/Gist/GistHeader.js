import React from "react";
import styled from "styled-components";
import IconLink from "./IconLink";

const getFilesLabel = (count) => {
  if (count == 0) return count + " " + "Files";
  if (count == 1) return count + " " + "File";
  if (count > 1) return count + " " + "Files";
};

const GistHeader = ({
  avatarURL,
  login,
  loginURL,
  commentsURL,
  forksURL,
  fileCount,
}) => {
  return (
    <Header>
      <Person>
        <Photo src={avatarURL} />
        <NameLink href={loginURL}>{login}</NameLink>
      </Person>

      <Row>
        <IconLink
          label={getFilesLabel(fileCount)}
          icon="code"
          style={iconLink}
        />
        <IconLink
          label="Forks"
          icon="repo-forked"
          link={forksURL}
          style={iconLink}
        />
        <IconLink
          label="Comments"
          icon="comment"
          link={commentsURL}
          style={iconLink}
        />
        <IconLink label="Stars" icon="star" />
      </Row>
    </Header>
  );
};

const Header = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  padding-bottom: 16px;
`;

const Person = styled.div`
  display: flex;
  align-items: center;
`;

const Photo = styled.img`
  width: 36px;
  height: 36px;
  border-radius: 500px;
`;

const NameLink = styled.a`
  font-size: 16px;
  color: ${(props) => props.theme.primary};
  text-decoration: none;
  cursor: pointer;
  margin-left: 8px;
`;

const Row = styled.div`
  display: flex;
  align-items: center;
`;

const iconLink = { marginRight: 16 };

export default GistHeader;
