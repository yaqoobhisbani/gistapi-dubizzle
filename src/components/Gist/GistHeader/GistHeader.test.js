import React from "react";
import { render } from "@testing-library/react";
import GistHeader from "./GistHeader";

describe("GistHeader", () => {
  test("matches snapshot", () => {
    const { container } = render(
      <GistHeader
        avatarURL="http://localhost:3000/avatar.png"
        commentsURL="http://localhost:3000/comments"
        login="yaqoobhisbani"
        loginURL="https://github.com/yaqoobhisbani"
        fileCount={3}
        forksURL="http://localhost:3000/forks"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
