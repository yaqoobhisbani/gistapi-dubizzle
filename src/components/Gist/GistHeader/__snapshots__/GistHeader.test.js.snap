// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GistHeader matches snapshot 1`] = `
<div>
  <div
    class="sc-dmqHEX fEozdp"
  >
    <div
      class="sc-hLseeU gVCbip"
    >
      <img
        class="sc-eDDNvR dbSxja"
        src="http://localhost:3000/avatar.png"
      />
      <a
        class="sc-jTrPJq dNVFCQ"
        href="https://github.com/yaqoobhisbani"
      >
        yaqoobhisbani
      </a>
    </div>
    <div
      class="sc-gLDzan cEgHXJ"
    >
      <div
        class="sc-beqWaB juKUET"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-code"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-gueYoa fnOQUP"
        >
          3 Files
        </a>
      </div>
      <div
        class="sc-beqWaB juKUET"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-repo-forked"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-gueYoa fnOQUP"
          href="http://localhost:3000/forks"
        >
          Forks
        </a>
      </div>
      <div
        class="sc-beqWaB juKUET"
        style="margin-right: 16px;"
      >
        <span
          class="octicon octicon-comment"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-gueYoa fnOQUP"
          href="http://localhost:3000/comments"
        >
          Comments
        </a>
      </div>
      <div
        class="sc-beqWaB juKUET"
      >
        <span
          class="octicon octicon-star"
          data-testid="icon"
          style="color: rgb(63, 132, 228); font-size: 16px; font-weight: 500;"
        />
        <a
          class="sc-gueYoa fnOQUP"
        >
          Stars
        </a>
      </div>
    </div>
  </div>
</div>
`;
